### PR TITLE
feat(evm): OpCodesHooks for CREATE and CALL opcodes

### PR DESCRIPTION
**Pull request by facs95**

### CODE REVIEW

File: core/vm/evm.go [hunk 1]
```diff
@@ -97,28 +97,54 @@ type EVM struct {
 	// activePrecompiles defines the precompiles that are currently active
 	activePrecompiles []common.Address
 
-	// preExecuteCallback is a callback function that is called before executing
-	// CALL, CALLCODE, DELEGATECALL and STATICCALL opcodes.
-	preExecuteCallback preExecuteCallbackType
+	// // preExecuteCallback is a callback function that is called before executing
+	// // CALL, CALLCODE, DELEGATECALL and STATICCALL opcodes.
+	// preExecuteCallback preExecuteCallbackType
```
Comment (0xstepit) [inline core/vm/evm.go]:
```suggestion
```

File: core/vm/evm.go [hunk 2]
```diff
@@ -97,28 +97,27 @@ type EVM struct {
...
-		chainConfig:        chainConfig,
-		chainRules:         chainConfig.Rules(blockCtx.BlockNumber, blockCtx.Random != nil),
-		preExecuteCallback: dummyCallback,
+		Context:     blockCtx,
+		TxContext:   txCtx,
+		StateDB:     statedb,
+		Config:      config,
+		chainConfig: chainConfig,
+		chainRules:  chainConfig.Rules(blockCtx.BlockNumber, blockCtx.Random != nil),
+		// preExecuteCallback: dummyCallback,
```
Comment (GAtom22) [inline core/vm/evm.go]:
```suggestion
```

File: core/vm/evm.go [hunk 3]
```diff
@@ -128,17 +127,17 @@ func NewEVM(blockCtx BlockContext, txCtx TxContext, statedb StateDB, chainConfig
 	return evm
 }
 
-// NewEVMWithCallback returns a new EVM and takes a custom preExecuteCallback. The returned EVM is
+// NewEVMWithHooks returns a new EVM and takes a custom preExecuteCallback. The returned EVM is
```
Comment (GAtom22) [inline core/vm/evm.go]:
```suggestion
// NewEVMWithHooks returns a new EVM and takes a custom OpCodeHooks. The returned EVM is
```

**Review by GAtom22 — APPROVED** [review]
LGTM! Great work @facs95!!

**Review by 0xstepit — APPROVED** [review]
great work!

**Review by ramacarlucho — APPROVED** [review]

**Pull request merged**